File: 80_Remove_Duplicate_From_Sorted_Array_II.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        k = len(nums)
        i = len(nums) - 2 #starting at the second to last element
        while (i >=  1): #Go to the second element
            #check current, previous, and next
            #if all are the same, remove current.
            if (nums[i] == nums[i - 1] and nums[i] == nums[i + 1]):
                nums.pop(i)
                k -= 1
            i -= 1
        return k

File: solutions/80_Remove_Duplicate_From_Sorted_Array_II.cpp
class Solution
{
public:
    int removeDuplicates(vector<int> &nums)
    {
        int k = nums.size();
        int i = nums.size() - 2; // start iterating on second to last element
        while (i >= 1)
        { // iterate until the second element
            // if previous, current, and next are all the same
            //  erase current
            if (nums[i] == nums[i - 1] && nums[i] == nums[i + 1])
            {
                nums.erase(nums.begin() + i);
                k--;
            }
            i--;
        }
        return k;
    }
};